### DISCUSSION TITLE
Problems detecting pos

### DISCUSSION THREAD

**Discussion by JessitaMS** [explosion/spaCy#9553] [Help: Coding & Implementations] [answered]
Hello,

I am new to using NLP. I have just tried passing a text to determine the POS and I think there is something that it detects wrong, but I do not understand why and how to solve it, if you can help me ...

![image](https://user-images.githubusercontent.com/26332548/139085631-5d14183a-1f6f-4452-82f0-edb817fddd08.png)

As you can see in the image, she understands perfectly that Elizabeth is a personal pronoun, but it strikes me that she walked around that is that a verb also understands it as another personal pronoun.

How could this recognition be arranged

Thanks

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @JessitaMS , thanks for the report! This is definitely not ideal, yet it happens due to the nature of most machine learning models. Typically, we collect these reports in a master thread #3052 . You might find something related there too

Although you can further improve your results by (1) finetuning the spaCy model to your own domain [[example1](https://github.com/explosion/projects/tree/v3/pipelines/tagger_parser_ud)] [[example2](https://github.com/explosion/projects/tree/v3/tutorials/ner_drugs)], (2) trying a larger [Spanish model]( https://spacy.io/models/es) (e.g.  `es_core_web_lg` or `es_dep_news_trf`) or (3) constructing a set of comprehensive rules to cover obvious tags.

**Reply by JessitaMS** [reply]
Hello @ljvmiranda921, I have tried to use the long Spanish model and it improves a little but still not enough, because although it is true that in the concrete example that I send you in the image it modifies the PROPN of "stroll" well, but in another certain part of the text the same thing happens to me again. With which this option does not serve me much.

On the other hand, I have built certain comprehension rules (through certain patterns) but although it works a little better for me, I still find comprehension difficulties. For example, when a character has a proper name, but in the following sentence it is already referred to as her / him. I don't know how to associate this with that same character. And another example of what happens to me is that when, for example, I try to extract the adjectives belonging to that subject, because I have no way of precisely associating the adjective with the subject, since on many occasions it does not follow the pattern of PROPN + ADJ and / or obj + amod.

With regard to adjusting the spacy to the domain with the work, I do not find clear the examples that it indicates to me. I guess because I am very new to using this algorithm.

If you could explain or indicate some reference to me where the way to adjust spacy to my domain is more understandable. Thanks!

**Reply by ljvmiranda921** [reply]
Hi @JessitaMS ,

> For example, when a character has a proper name, but in the following sentence it is already referred to as her / him. I don't know how to associate this with that same character

That's what we usually call a [coreference](https://en.wikipedia.org/wiki/Coreference) problem--whenever we want to find two (or more) expressions that refer to the same thing (e.g. a pronoun and a proper noun). Unfortunately we don't have a working coref model for this. You can explore tools like [corefree](https://spacy.io/universe/project/coreferee) to solve that 

>  I try to extract the adjectives belonging to that subject, because I have no way of precisely associating the adjective with the subject, since on many occasions it does not follow the pattern of PROPN + ADJ and / or obj + amod.

You can check the [DependencyMatcher](https://spacy.io/api/dependencymatcher) to look for adjectives 

> If you could explain or indicate some reference to me where the way to adjust spacy to my domain is more understandable.

Hmm, taking back what I said earlier. Perhaps a better way to address this problem is to fix mistakes as you find them using the [AttributeRuler](https://spacy.io/api/attributeruler) or creating a custom component. If you find *a lot* of mistakes that seem to follow a particular pattern, then we may be able to fix it. But fixing a  very specific case by improving the model is nontrivial (cf. #3052).
